src/main/scala/com/fulcrumgenomics/sv/tools/AggregateSvPileup.scala

```diff
@@ -0,0 +1,495 @@
+        Some(contigToTargets.toMap)
+    }
+
+    // Open output writer
+    val writer = Metric.writer[AggregatedBreakpointPileup](output)
+
+    // Group pileups by left and right contig and strand
+    pileups.groupMap(BreakpointContigOrientation(_))(identity).foreach { case (_, group) =>
+
+      // Within each contig+strand group, identify all immediate "neighbors" of each pileup (other pileups within the
+      // distance threshold)
+      val pileupToNeighbors: Map[BreakpointPileup, PileupGroup] = AggregateSvPileup.pileupToNeighbors(group, maxDist)
+
+      // Within each contig+strand group, recursively build (transitive) clusters of neighboring pileups
+      val pileupClusters: Seq[PileupGroup] = AggregateSvPileup.toClusters(pileupToNeighbors)
+      logger.info(f"Processing ${pileupClusters.size} pileup clusters with endpoints " +
+        f"${group.head.left_contig}:${group.head.left_strand}, ${group.head.right_contig}:${group.head.right_strand}")
+
+      // Aggregate each cluster, analyze allele frequency and target overlap, and write output
+      pileupClusters
+        .map(AggregatedBreakpointPileup.apply)
+        .map(_.calculateFrequency(samSource, flank, minTotal, minFrequency))
+        .map(_.addTargetOverlap(targets))
+        .foreach(writer.write)
+    }
+
+    writer.close()
+
+  }
+
+}
+
+
+object AggregateSvPileup {
+
+  /** Type alias for a collection of pileups */
+  type PileupGroup = Seq[BreakpointPileup]
+
+  /**
+   * Maps each pileup to the set of other pileups within the distance threshold.
+   *
+   * Note: it is recommended to call this method on a group of pileups that share a common `BreakpointContigOrientation`
+   * to reduce unnecessary run time.
+   *
+   * @param pileups Set of pileups to evaluate for pairwise neighbor relationships
+   * @param maxDist Maximum distance (inclusive) for the left and right coordinates to be considered a neighbor
+   * @return Map of pileup to the set of neighbors
+   */
+  private def pileupToNeighbors(pileups: PileupGroup, maxDist: Int): Map[BreakpointPileup, PileupGroup] = {
+    val map: mutable.Map[BreakpointPileup, List[BreakpointPileup]] = mutable.Map.from(pileups.map((_, Nil)))
+    for (pileup1 <- pileups; pileup2 <- pileups) {
+      if (pileup1.total >= pileup2.total && isCluster(pileup1, pileup2, maxDist)) {
+        map.put(pileup1, (pileup2 :: map(pileup1)).distinct)
+        map.put(pileup2, (pileup1 :: map(pileup2)).distinct)
+      }
+    }
+    map.toMap
+  }
+
+  /** Returns true if the two breakpoints can be clustered, i.e., they are not the same breakpoint (have different
+   * IDs), they have the same left and right chromosomes and strand, and the distance between their left and right
+   * coordinates is below the threshold (inclusive) */
+  private def isCluster(bp1: BreakpointPileup, bp2: BreakpointPileup, maxDist: Int): Boolean = {
+    bp1.id != bp2.id &&
+      BreakpointContigOrientation(bp1) == BreakpointContigOrientation(bp2) &&
```

Comment:
There's an awful lot of creation of BreakpointContigOrientation objects here and elsewhere.  Perhaps this should just be a `val` or `lazy val` on BreakpointPileup` so we don't make it over and over?